moviebot/nlu/annotation/joint_bert/joint_bert.py

```diff
@@ -0,0 +1,97 @@
+"""Joint BERT model for intent classification and slot annotation."""
+import os
+from typing import List, Optional, Tuple
+
+import torch
+import torch.nn as nn
+from transformers import BertModel
+
+_BERT_BASE_MODEL = "bert-base-uncased"
+
+
+class JointBERT(nn.Module):
+    def __init__(
+        self,
+        intent_label_count: int,
+        slot_label_count: int,
+    ) -> None:
+        """Initializes the JointBERT model.
+
+        Args:
+            intent_label_count: The number of intent labels.
+            slot_label_count: The number of slot labels.
+        """
+        super(JointBERT, self).__init__()
+
+        self.slot_label_count = slot_label_count
+        self.intent_label_count = intent_label_count
+
+        self.bert = BertModel.from_pretrained(_BERT_BASE_MODEL)
+        self.intent_classifier = nn.Linear(
+            self.bert.config.hidden_size, intent_label_count
+        )
+        self.slot_classifier = nn.Linear(
+            self.bert.config.hidden_size, slot_label_count
+        )
+
+    def forward(
+        self, input_ids: torch.Tensor, attention_mask: torch.Tensor
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Forward pass of the model.
+
+        Args:
+            input_ids: The input token IDs.
+            attention_mask: The attention mask.
+
+        Returns:
+            The intent and slot logits.
+        """
+        outputs = self.bert(input_ids, attention_mask=attention_mask)
+        intent_logits = self.intent_classifier(outputs.pooler_output)
+        slot_logits = self.slot_classifier(outputs.last_hidden_state)
+        return intent_logits, slot_logits
+
+    def predict(
+        self,
+        input_ids: torch.Tensor,
+        attention_mask: Optional[torch.Tensor] = None,
+    ) -> Tuple[int, List[int]]:
+        """Predicts the intent and slot annotations for the given input.
+
+        Args:
+            input_ids: The input token IDs.
+            attention_mask (optional): The attention mask. Defaults to None.
+
+        Returns:
+            A tuple of the predicted intent and slot annotations.
+        """
+        if attention_mask is None:
```

Comment:
`if not attention_mask:`

Reply:
RuntimeError: Boolean value of Tensor with more than one value is ambiguous.

It has to be compared to None explicitly.